packages/react-client/src/ReactFlightClient.js

```diff
@@ -512,6 +512,27 @@ function wakeChunk<T>(
       fulfillReference(listener, value, chunk);
     }
   }
+
+  if (__DEV__ && chunk.status === INITIALIZED) {
```

Comment:
You can change the type to just accept an initialized chunk since all the callsites except the cycle resolution is doing that.

Then fork the cycle resolution callsite maybe.